### PR TITLE
skip all pipelines based on what is in the PR

### PR DESCRIPTION
**Pull request by matttrach**
#### Proposed Changes ####

This change will add a pipeline which implements [drone-skip-pipeline](https://github.com/joshdk/drone-skip-pipeline).
The purpose is to provide some mechanism to automatically skip pipelines which are not necessary due to the files that are being changed. For example, if a markdown file is the only file changed, then we don't need to run functional tests. This will reduce our integration time.

#### Types of Changes ####

No new code to k3s, all changes in .drone.yml

#### Verification ####

Unfortunately, Drone provides no way to test this outside of letting it run in your repo.

#### Linked Issues ####

This addresses #6980.

#### User-Facing Change ####
This has no user facing change.

#### Further Comments ####

There is [another plugin](https://github.com/fernandrone/drone-ignore-config) which may give us similar functionality, but is out of date and requires configuration on the drone server, I would like to try this one first to avoid the additional handoffs.
There is more than one way to implement this plugin, the way I have provided is the one with the fewest changes to existing steps, I am open to additional suggestions.

### CODE REVIEW

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Review by brandond — CHANGES_REQUESTED** [review]
I'm not sure what's going on with all the commits that have been added to this branch after approval; please get it back to just the single change before merging.

**Comment by matttrach** [comment]
sorry, I will. I was finally able to test this and it doesn't work... :(
The plugin is only compiled on amd64, I think I will need to head back to the drawing board on this one.

**Comment by matttrach** [comment]
I put everything into a single commit, I believe this is ready for review.

File: .drone.yml
Comment (matttrach) [inline .drone.yml:506]: the diff is unclear here, but if you look at the file I just removed some spacing here to conform to the way the rest of the yaml is written (lists have same spacing as parent)

File: README.md
Comment (matttrach) [inline README.md:164]: This is a trivial spacing change to validate that .md files are ignored.

**Review by cwayne18 — COMMENTED** [review]
This looks neat thanks! Can we add .github/*,  MAINTAINERS, CODEOWNERS to the skip list please?

**Comment by matttrach** [comment]
done. Also added DCO, channel.yaml, and LICENSE (although they are unlikely to change)

**Review by dereknola — APPROVED** [review]
Is is possible to put all the repeated bash commands into a script file and just call that?

**Comment by matttrach** [comment]
I think the special exit code has to be executed from the root scope, I can probably add a file and execute it, but it would only  reduce 1 line of code per copy.

**Review by cwayne18 — APPROVED** [review]

**Review by brandond — CHANGES_REQUESTED** [review]
Do we want this on all CI runs, or just on PR? It feels like we should never skip running CI on merges or tags - basically anything that runs on drone-publish. Should we add something to restrict this to just drone-pr?

```yaml
trigger:
  instance:
  - drone-pr.k3s.io
  ref:
  - refs/pull/*/head
```

**Comment by matttrach** [comment]
Definitely don't wanna skip on tags, thanks!
I think bounding the skip to pushes and pull requests should give us a good compromise for how often.

**Review by brandond — APPROVED** [review]

**Review by ivan-claire — APPROVED** [review]
Looks good to me.

**Pull request merged**
